### PR TITLE
Fix issue with Remove and Create function not incrementing the ID

### PR DESCRIPTION
**Pull request by underscore05**
## Description
- Fix issue with Create function not incrementing the id and just use the passed id.
- Fix issue with Remove function using Boolean return type but it actually uses the Type's name.

- Existing `remove`Typename has a return type of boolean but in reality it returns an object or undefined.
- Existing create function overrides the incremental id attribute and just use the id coming from the api.

## Related Issue
- https://github.com/marmelab/json-graphql-server/issues/43
- https://github.com/marmelab/json-graphql-server/issues/102

### CODE REVIEW

File: src/resolver/Mutation/create.spec.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import create from './create';
 
-test('returns a new object with id 0 on empty datastore', () => {
-    expect(create()(null, {})).toEqual({ id: 0 });
+test('returns a new object with id 1 on empty datastore', () => {
+    expect(create()(null, {})).toEqual({ id: 1 });
```
Comment (fzaninotto) [inline src/resolver/Mutation/create.spec.js:4]: This is a breaking change, why did you do it?

File: src/introspection/getSchemaFromData.js [hunk 2]
```diff
@@ -140,16 +140,25 @@ export default (data) => {
                 },
                 {}
             );
+            const createFields = Object.keys(typeFields).reduce(
+              (f, fieldName) => {
```
Comment (fzaninotto) [inline src/introspection/getSchemaFromData.js:144]: the `f` variable is cryptic. Perhaps rename it to `acc` for accumulator?

File: src/introspection/getSchemaFromData.spec.js [hunk 3]
```diff
@@ -228,7 +222,7 @@ test('creates three mutation fields per data type', () => {
             description: null,
         },
     ]);
-    expect(mutations['removePost'].type.name).toEqual(GraphQLBoolean.name);
+    expect(mutations['removePost'].type.name).toEqual(PostType.name);
```
Comment (fzaninotto) [inline src/introspection/getSchemaFromData.spec.js:225]: why this change?

File: src/introspection/getSchemaFromData.js [hunk 4]
```diff
@@ -140,16 +140,25 @@ export default (data) => {
...
-                args: typeFields,
+                args: createFields,
             };
             fields[`update${type.name}`] = {
                 type: typesByName[type.name],
                 args: nullableTypeFields,
             };
             fields[`remove${type.name}`] = {
-                type: GraphQLBoolean,
+                type: typesByName[type.name],
```
Comment (fzaninotto) [inline src/introspection/getSchemaFromData.js:161]: I don't understand this change

File: src/introspection/getSchemaFromData.js [hunk 5]
```diff
@@ -140,16 +140,25 @@ export default (data) => {
...
             );
+            const createFields = Object.keys(typeFields).reduce(
+              (f, fieldName) => {
+                if (fieldName !== 'id') {
+                  f[fieldName] = Object.assign({}, typeFields[fieldName])
+                }
+                return f
+              },
+              {}
+            );
```
Comment (fzaninotto) [inline src/introspection/getSchemaFromData.js:151]: I think there is a simpler way to do this:

```js
const { id, ...createFields } = typeFields;
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/introspection/getSchemaFromData.js (hunk 4 above)
Comment (underscore05) [reply src/introspection/getSchemaFromData.js:161]: The actual implementation of `remove` returns an object instead of Boolean causing an error in delete call. So, instead of changing it I just change the return type so it will match.

File: src/resolver/Mutation/create.spec.js (hunk 1 above)
Comment (underscore05) [reply src/resolver/Mutation/create.spec.js]: Already reverted the initial generated id to 0

File: src/introspection/getSchemaFromData.js (hunk 2 above)
Comment (underscore05) [reply src/introspection/getSchemaFromData.js]: Already applied to object destructuring approach.

File: src/resolver/Mutation/create.js [hunk 6]
```diff
@@ -1,8 +1,7 @@
 export default (entityData = []) => (_, entity) => {
     const newId =
         entityData.length > 0 ? entityData[entityData.length - 1].id + 1 : 0;
-    const newEntity = Object.assign({ id: newId }, entity);
-
+    const newEntity = Object.assign(entity, { id: newId });
```
Comment (fzaninotto) [inline src/resolver/Mutation/create.js:4]: Object.assign mutates the first argument. The correct syntax should be:

```js
Object.assign({}, entity, { id: newId });
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: src/resolver/Mutation/create.js (hunk 6 above)
Comment (underscore05) [reply src/resolver/Mutation/create.js]: Hello @fzaninotto already updated my PR

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!
